Loop/Managers/UserAlert/InAppUserAlertHandler.swift

```diff
@@ -0,0 +1,113 @@
+//
+
+import Foundation
+import LoopKit
+
+public class InAppUserAlertHandler: UserAlertHandler {
+    
+    private weak var rootViewController: UIViewController?
+    
+    private var alertsShowing: [(UIAlertController, UserAlert)] = []
+    private var alertsPending: [(Timer, UserAlert)] = []
+    
+    public init(rootViewController: UIViewController) {
+        self.rootViewController = rootViewController
+    }
+        
+    public func scheduleAlert(_ alert: UserAlert) {
+        switch alert.trigger {
+        case .immediate:
+            show(alert: alert)
+        case .delayed(let interval):
+            schedule(alert: alert, interval: interval, repeats: false)
+        case .repeating(let interval):
+            schedule(alert: alert, interval: interval, repeats: true)
+        }
+    }
+    
+    public func unscheduleAlert(managerIdentifier: String, typeIdentifier: UserAlert.TypeIdentifier) {
+        DispatchQueue.main.async {
+            self.alertsPending.filter {
+                $0.1.identifier == UserAlert.getIdentifier(managerIdentifier: managerIdentifier, typeIdentifier: typeIdentifier)
+            }
+            .forEach { timer, alert in
+                timer.invalidate()
+            }
+        }
+    }
+    
+    public func cancelAlert(managerIdentifier: String, typeIdentifier: UserAlert.TypeIdentifier) {
+        DispatchQueue.main.async {
+            self.alertsShowing.filter {
+                $0.1.identifier == UserAlert.getIdentifier(managerIdentifier: managerIdentifier, typeIdentifier: typeIdentifier)
+            }
+            .forEach { alertController, alert in
+                alertController.dismiss(animated: true)
+            }
+        }
+    }
+}
+
+/// Private functions
+extension InAppUserAlertHandler {
+    
+    private func schedule(alert: UserAlert, interval: TimeInterval, repeats: Bool) {
+        guard alert.foregroundContent != nil else {
+            return
+        }
+        DispatchQueue.main.async {
+            let timer = Timer.scheduledTimer(withTimeInterval: interval, repeats: repeats) { [weak self] timer in
+                self?.show(alert: alert)
+                if !repeats {
+                    self?.alertsPending.removeAll { $0.0 == timer && $0.1.identifier == alert.identifier }
+                }
+            }
+            self.alertsPending.append((timer, alert))
+        }
+    }
+    
+    private func show(alert: UserAlert) {
+        guard let content = alert.foregroundContent else {
+            return
+        }
+        DispatchQueue.main.async {
+            guard self.alertsShowing.contains(where: { $1.identifier == alert.identifier }) == false else {
+                return
+            }
+            let alertController = self.presentAlert(title: content.title, message: content.body, action: content.acknowledgeAction) {
+                self.alertsShowing.removeAll { $1.identifier == alert.identifier }
```

Comment:
Just looked through the other PR and now I'm not sure what the `identifier` means. See comments there which will obviously feed into comments here.

Reply:
The reality is it will be very rare for there to be more than one alert showing.  Rare...but not impossible, hence the array.

`identifier` is a tuple of `( managerIdentifier, typeIdentifier)`, in order to uniquely identify an alert.  You're right, I put comments in `UserAlert.swift` for `managerIdentifier` and `typeIdentifier` but not `identifier`.  I'll make that clearer, thanks!

Reply:
Follow-up question, can the same type of alert exist in the array multiple times?

Reply:
No, the same alert should not exist in the array multiple times.

Reply:
then why not use a dictionary to ensure this doesn't happen and also make the searching more efficient?

Reply:
It isn't necessary.  And, though I'd claim it would be a case of premature optimization, it still turns out dictionary lookup would actually be _slower_.  For arrays of less than about 7-10, iterating through an array is faster than a hash value evaluation.

Reply:
It will definitely help ensuring there is only 1 alert type in the stack. Also, isn't a dictionary O(1) operation while an unordered array is O(n)?

Reply:
You are correct as to the functional order (O-notation), but I encourage you to try it yourself: Try timing a dictionary lookup of a reasonably-sized string hash value calculation vs. an array of strings less than 7.
I'll change it by your request, but I still think I'm right :)